Add links between Azure protection guides

ptgott opened the pull request:
Closes #54091

There are two guides for protecting Azure with Teleport, and each uses a different set of Teleport features. Add links between the two guides.

ptgott committed:
Add links between Azure protection guides

Closes #54091

There are two guides for protecting Azure with Teleport, and each uses a
different set of Teleport features. Add links between the two guides.

flyinghermit commented on docs/pages/enroll-resources/application-access/cloud-apis/azure.mdx:
```diff
@@ -8,9 +8,14 @@ description: How to enable secure access to Azure CLIs.
 
+If you have enabled the Teleport SAML IdP, you can use it instead of the
+Teleport Application Service to manage access to Azure CLI applications. For
+instructions, read [Access Azure Portal and
```
Once you update SAML docs, update this copy with similarities/differences too.

flyinghermit reviewed (commented):
Thanks for this PR. 
I have added a few suggestions to call out similarities and differences between two methods. Otherwise, it will be confusing for customers to decide which method to use over another.

ptgott commented on docs/pages/enroll-resources/application-access/cloud-apis/azure.mdx:
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Protect Azure CLIs with Teleport Application Access"
+title: Protect Azure CLIs and APIs with Teleport
```
We phased out the "[Resource] Access" naming convention after we stopped positioning protocols as individual products and combined them all under a single product, which is now called Teleport Zero Trust Access.

ptgott committed:
Respond to flyinghermit feedback

Provide more nuance in describing the two available integration
approaches. This includes adding a "How it works" section to the Azure
Portal guide, fulfilling one expectation we are standardizing for how-to
guides in general.

ptgott commented:
@flyinghermit I'll keep this PR up to date until you are available to push commits to correct and expand the information it adds. Thanks!

flyinghermit commented:
Sorry @ptgott, I got occupied with some other priority tasks and could not look into this. I will try to get to it next week when I will be jumping back on docs for couple updates I have been tagged already.

P.S don't worry on keeping the PR up to date. I will sync with master at once.

flyinghermit committed:
updte how it works

flyinghermit commented:
@ptgott Apologies for the delay here. I have just updated the "How it works" section in https://github.com/gravitational/teleport/pull/54315/commits/dbc926a9dcde45aef0185491b354b649a47836a1. Please give it a review and feel free to apply preferred docs style magic as needed.

ptgott commented:
> @ptgott Apologies for the delay here. I have just updated the "How it works" section in [dbc926a](https://github.com/gravitational/teleport/commit/dbc926a9dcde45aef0185491b354b649a47836a1). Please give it a review and feel free to apply preferred docs style magic as needed.

Thanks! Left some minor grammatical suggestions in the commit view.

ptgott committed:
Apply minor grammar and style changes

ptgott commented:
@flyinghermit Thanks for making changes! I've applied some minor grammatical/style tweaks in https://github.com/gravitational/teleport/pull/54315/commits/07763269585a018af13a959bfa24fdc161c641b7.

flyinghermit reviewed (approved):
Thank you @ptgott

ptgott commented:
@greedy52 Ah, good point, thanks! @flyinghermit Would you like to take this one on as well?

ptgott commented:
@greedy52 Since this PR has been open for so long, I am leaning toward getting it merged without editing the AKS guide, which we can modify separately.
